assets/scripts/dialogs/SignInDialog.jsx

```diff
@@ -40,6 +43,15 @@ export default class SignInDialog extends React.Component {
     })
   }
 
+  handleUniversalSignIn = (event) => {
+    this.setState({
+      signingIn: true
+    })
+
+    const locale = store.getState().locale.locale
```

Comment:
@louh I wanted to draw attention to this. I _think_ it's the right way to snag the locale? If not, let me know.

Reply:
Yes, that looks right!

There is a "more correct" way to do this in React components: instead of importing `store` and calling `store.getState()`, you use the `connect()` function from `react-redux`. Here's an example of this in use: https://github.com/streetmix/streetmix/blob/main/assets/scripts/app/StreetView.jsx#L374-L382

Docs on this, just in case you need it: https://react-redux.js.org/tutorials/connect

To be consistent with other React components I'd recommend using this syntax. I don't think it's _wrong_ to import `store` this way, but we export it as a convenience for non-React parts of the app; I'd prefer to keep the React parts using the "standard" React way of connecting state.

Reply:
Super, thank you! I'll fix that.

Reply:
@louh Would it be Bad if we plucked the locale from the Redux store in the routing.js file instead ([right here](https://github.com/streetmix/streetmix/pull/2329/files#diff-1c67a9ed013a3acb17c7872da9f240e6fc7ef6fe485c8689d353ca36444c56fbR62))? We're never over-riding it between pulling from store and passing to `goUniversalSignIn()`, so making it a param is really just so routing.js doesn't call the store. If that's not a concern, I'd bury it in there instead.